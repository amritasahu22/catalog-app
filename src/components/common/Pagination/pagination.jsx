import React from "react";
import PropTypes from "prop-types";
import { range } from "lodash";

const Pagination = (props) => {
	const {
		totalItems,
		pageSize,
		currentPage,
		totalPageCount,
		onPageChange,
	} = props;

	const pages = range(1, totalPageCount + 1);

	return (
		<nav className="Paginate">
			<ul
				className="pagination pagination-md
			  justify-content-end clickable"
			>
				<li className={currentPage === 1 ? "page-item disabled" : "page-item"}>
					<a
						className="page-link text-dark bg-light"
						aria-label="Previous"
						onClick={() => onPageChange(currentPage - 1)}
					>
						<span aria-hidden="true">&laquo;</span>
						<span className="sr-only">Previous</span>
					</a>
				</li>

				{pages.map((page) => {
					if (
						page === 1 ||
						(page >= currentPage - 2 && page <= currentPage + 2)
					) {
						return (
							<li
								className={
									page === currentPage ? "page-item active" : "page-item"
								}
								key={page}
							>
								<a
									className="page-link text-dark bg-light"
									onClick={() => onPageChange(page)}
								>
									{page}
								</a>
							</li>
						);
					}
				})}

				<li
					className={
						currentPage === totalPageCount ? "page-item disabled" : "page-item"
					}
				>
					<a
						className="page-link text-dark bg-light"
						aria-label="Next"
						onClick={() => onPageChange(currentPage + 1)}
					>
						<span aria-hidden="true">&raquo;</span>
						<span className="sr-only">Next</span>
					</a>
				</li>
			</ul>
		</nav>
	);
};

Pagination.propTypes = {
	totalItems: PropTypes.number.isRequired,
	pageSize: PropTypes.number.isRequired,
	currentPage: PropTypes.number.isRequired,
	totalPageCount: PropTypes.number,
	onPageChange: PropTypes.func.isRequired,
};

export default Pagination;
