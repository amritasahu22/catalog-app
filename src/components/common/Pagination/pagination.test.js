import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Pagination from "./Pagination";

describe("Pagination", () => {
	test("should render properly with props", () => {
		const component = shallow(
			<Pagination
				totalItems={50}
				pageSize={10}
				currentPage={1}
				totalPageCount={5}
				onPageChange={() => {}}
			/>
		);
		expect(toJson(component)).toMatchSnapshot();
	});
});
