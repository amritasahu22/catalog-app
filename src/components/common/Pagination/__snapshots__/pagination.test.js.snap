// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render properly with props 1`] = `
<nav
  className="Paginate"
>
  <ul
    className="pagination pagination-md
 justify-content-end clickable"
  >
    <li
      className="page-item disabled"
    >
      <a
        aria-label="Previous"
        className="page-link text-dark bg-light"
        onClick={[Function]}
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
    </li>
    <li
      className="page-item active"
      key="1"
    >
      <a
        className="page-link text-dark bg-light"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="page-item"
      key="2"
    >
      <a
        className="page-link text-dark bg-light"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="page-item"
      key="3"
    >
      <a
        className="page-link text-dark bg-light"
        onClick={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        aria-label="Next"
        className="page-link text-dark bg-light"
        onClick={[Function]}
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
